### PR TITLE
install: Print eol warnings before starting to install

### PR DESCRIPTION
**Pull request by matthiasclasen**
We should inform users that the software they are about to install is end-of-life and won't get updates. And give them a chance to opt out.

### CODE REVIEW

File: app/flatpak-builtins-utils.c [hunk 1]
```diff
@@ -1088,3 +1088,50 @@ as_store_find_app (AsStore *store,
...
+       if (col + space + len >= cols)
+         {
+           g_print ("\n%s", words[i]);
+           col = len;
+         }
+       else
+         {
+           g_print ("%*s%s", space, "", words[i]);
+           col = col + space + len;
+         }
```
Comment (alexlarsson) [inline app/flatpak-builtins-utils.c]: This doesn't seem to correctly handle the case where the first word is wider than cols. We should always print at least one word per line.

**Comment by alexlarsson** [comment]
Don't we already print these in the end_of_lifed() signal callback?

**Comment by matthiasclasen** [comment]
> Don't we already print these in the end_of_lifed() signal callback?

Hmm, you're right we do. But I think it makes more sense to show them before the user hits the big "GO" button, and give him a chance to opt out. Maybe we don't need the end_of_lifed handler anymore

**Comment by matthiasclasen** [comment]
good testcase: epiphany on flathub

**Comment by alexlarsson** [comment]
So, i'm not sure about this one. This uses the "global remote state" for the eol data, whereas the signal actually uses the data from the exact commit that we're pulling. This means for instance if you revert to an older version or something we're going to display the wrong data.

**Comment by matthiasclasen** [comment]
ah ok. needs more thought then. leave it for now

**Comment by alexlarsson** [comment]
This is handled by https://github.com/flatpak/flatpak/pull/2775, closing this

**Pull request closed without merging**